core/primitives/rle_bitset/rle_bitset.hpp

```diff
@@ -25,27 +25,46 @@ namespace fc::primitives {
       insert(other.begin(), other.end());
     }
 
+    inline void operator+=(const std::vector<RleBitset> &others) {
+      for (const auto &other : others) {
+        insert(other.begin(), other.end());
+      }
+    }
+
     inline RleBitset operator+(const RleBitset &other) const {
       auto result{*this};
       result += other;
       return result;
     }
 
-    inline RleBitset operator-(const RleBitset &other) const {
-      RleBitset result;
-      for (auto i : *this) {
-        if (!other.has(i)) {
-          result.insert(i);
+    inline RleBitset operator+(const std::vector<RleBitset> &others) const {
+      auto result{*this};
+      for (const auto &other : others) {
+        result += other;
+      }
+      return result;
+    }
+
+    inline void operator-=(const RleBitset &other) {
```

Comment:
both `this` and `other` are sorted.
if needed, this method can be optimized to single pass loop without `find`.